indico/modules/categories/models/event_move_request.py

```diff
@@ -0,0 +1,114 @@
+# This file is part of Indico.
+# Copyright (C) 2002 - 2021 CERN
+#
+# Indico is free software; you can redistribute it and/or
+# modify it under the terms of the MIT License; see the
+# LICENSE file for more details.
+
+from indico.core.db import db
+from indico.core.db.sqlalchemy import PyIntEnum, UTCDateTime
+from indico.modules.events import EventLogKind, EventLogRealm
+from indico.util.date_time import now_utc
+from indico.util.enum import RichIntEnum
+from indico.util.locators import locator_property
+
+
+class MoveRequestState(RichIntEnum):
+    pending = 0
+    accepted = 1
+    rejected = 2
+    withdrawn = 3
+
+
+class EventMoveRequest(db.Model):
+    """A request that represents a proposed event move."""
+
+    __tablename__ = 'event_move_requests'
+    __table_args__ = (db.Index(None, 'event_id', unique=True,
+                               postgresql_where=db.text(f'state = {MoveRequestState.pending}')),
+                      db.CheckConstraint(f'state in ({MoveRequestState.accepted}, {MoveRequestState.rejected}) '
+                                         f'AND moderator_id IS NOT NULL OR moderator_id IS NULL',
+                                         'moderator_state'),
+                      {'schema': 'categories'})
+
+    id = db.Column(
+        db.Integer,
+        primary_key=True
+    )
+    event_id = db.Column(
+        db.Integer,
+        db.ForeignKey('events.events.id'),
+        nullable=False,
+        index=True
+    )
+    category_id = db.Column(
+        db.Integer,
+        db.ForeignKey('categories.categories.id'),
+        nullable=True,
+        index=True
+    )
+    submitter_id = db.Column(
+        db.Integer,
+        db.ForeignKey('users.users.id'),
+        nullable=True,
+        index=True
+    )
+    state = db.Column(
+        PyIntEnum(MoveRequestState),
+        default=MoveRequestState.pending,
+        nullable=False
+    )
+    moderator_id = db.Column(
+        db.Integer,
+        db.ForeignKey('users.users.id'),
+        nullable=True
+    )
+    submitted_dt = db.Column(
+        UTCDateTime,
+        nullable=False,
+        default=now_utc
+    )
+
+    event = db.relationship(
+        'Event',
+        lazy=True,
+        backref=db.backref(
+            'move_requests',
+            cascade='all, delete-orphan',
+            lazy='dynamic'
+        )
+    )
+    category = db.relationship(
+        'Category',
+        lazy=True,
+        backref=db.backref(
+            'event_move_requests',
+            cascade='all, delete-orphan',
+            lazy='dynamic'
+        )
+    )
+    submitter = db.relationship(
+        'User',
+        lazy=True,
+        foreign_keys=submitter_id,
+        backref=db.backref(
+            'event_move_requests',
+            cascade_backrefs=False,
```

Comment:
I don't think this is needed/useful here? We only use if for a few places in EventPersons so we can create an EventPerson linked to a user but do not cascade adding it to the session unless it gets explicitly added.

Reply:
Yes, it doesn't seem needed.